### DISCUSSION TITLE
gh-pages deployment runs every time I push

### DISCUSSION THREAD

**Discussion by vibhav950** [community/community#143990] [Repositories]
I have set up a deployment to publish a site from the `/docs` folder of the `main` branch. However, it deploys the site every time I push to the main branch. Is there some way to configure the deployment pipeline so that the site only gets republished if the source files in `/docs` change?

**Comment by uncenter** [comment]
See https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-including-paths.

**Comment by Anonav0** [comment]
It is the default behavior of pages, so it'll happen nonetheless. And if you are new to this space , i'll give you a simple solution to follow.

step 1: create another branch named like `production`
step 2: commit your changes to it
step 3: push the branch to github.
step 4: configure the repo and set the deploying branch to `production`.

This will have make help you 'take control' of the deployments. Just make sure to update the `production` branch when ever you want to deploy your desired code.

Hope it works.

**Reply by uncenter** [reply]
This isn't really answering the question.

**Reply by Anonav0** [reply]
oh right. I'm sorry I didn't noticed the last line properly. Should I keep the answer or should I delete it?
